### DISCUSSION TITLE
linked device, but can't send messages

### DISCUSSION THREAD

**Discussion by sinback** [AsamK/signal-cli#436] [Q&A]
If I register the device as the master for my phone number, I can send messages, but if I link the device as a slave to my Android device (which is the use case I prefer), I can't send messages. The CLI reports: "Authorization failed, was the number registered elsewhere?
Failed to send message: Authorization failed!" -- is sending messages supported for devices which are linked to master devices?

**Comment by Trolldemorted** [comment]
Sending messages for slaves should definitely work. Sounds like the linking process didn't work?

**Comment by AsamK** [comment]
Does receiving work? Authorization failed is normally global, so if sending fails with that error, nothing else should work either.

what's the output
`signal-cli -u +NUMBER listDevices`

**Comment by riazm** [comment]
I have the same thing, listDevices returns 

`Authorization failed, was the number registered elsewhere?
org.whispersystems.signalservice.api.push.exceptions.AuthorizationFailedException: Authorization failed!
	at org.whispersystems.signalservice.internal.push.PushServiceSocket.makeServiceRequest(PushServiceSocket.java:699)
	at org.whispersystems.signalservice.internal.push.PushServiceSocket.getDevices(PushServiceSocket.java:165)
	at org.whispersystems.signalservice.api.SignalServiceAccountManager.getDevices(SignalServiceAccountManager.java:345)
	at org.asamk.signal.Manager.getLinkedDevices(Manager.java:427)
	at org.asamk.signal.Main.handleCommands(Main.java:317)
	at org.asamk.signal.Main.main(Main.java:84)
`
On my phone the device appears to be linked

**Comment by gxmobile** [comment]
I encountered this exact same error after linking an Android device. Removing/unregistering the device  did not help. I was finally able to resolve the issue by using `updateAccount` after linking the device. The full command to do this was:

    signal-cli -u USERNAME updateAccount

After that I checked `listDevices` again and no longer received an error. Everything has worked perfectly from then on.

**Comment by ghost** [comment]
I followed the @gxmobile's steps to no avail

**Comment by hoffmaan** [comment]
Hi, I can't send Messages after a fresh install (Raspbian 10, OpenJDK 11.0.8, signal-cli 0.6.10). This is what happens:

```
pi@pi01:~ $ sudo -u fhem signal-cli -u +Sender register --voice
OpenJDK Server VM warning: You have loaded library /tmp/resource10430407282192776788.so which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
pi@pi01:~ $
```

I DID receive a call with a verification code though, so I assumed I could ignore these complaints about a disabled stack guard for now. But obviously something went wrong (or has been wrong from the beginning). All commands result in the same (or very similar) messages:

```
pi@pi01:~ $ sudo -u fhem signal-cli -u +Sender verify Code
OpenJDK Server VM warning: You have loaded library /tmp/resource10430281644673653797.so which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
pi@pi01:~ $
```

```
pi@pi01:~ $ signal-cli -u +Sender send -m "Greetings from FHEM!" +Receiver
OpenJDK Server VM warning: You have loaded library /tmp/resource15414818810970183832.so which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
Failed to get sender certificate: org.whispersystems.signalservice.api.push.exceptions.AuthorizationFailedException: Authorization failed!
Failed to send message: Authorization failed!
pi@pi01:~ $
```

```
pi@pi01:~ $ signal-cli -u +Sender listDevices
OpenJDK Server VM warning: You have loaded library /tmp/resource8347196063280918122.so which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
User is not registered.
pi@pi01:~ $
```

```
pi@pi01:~ $ signal-cli -u +4960819857232 updateAccount
OpenJDK Server VM warning: You have loaded library /tmp/resource9321971295331710016.so which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
User is not registered.
pi@pi01:~ $
```

Any idea what happened and how I can possibly fix it? (Just to have it mentioned, I'm not a Java expert.)

Thanks for any help.

**Comment by ferristseng** [comment]
I was able to resolve this by removing my local data file, and re-linking my device just FYI:

```
rm ~/.local/share/signal-cli/data/+Sender
```

**Comment by hoffmaan** [comment]
Thank you for your suggestion. Didn't work for me, though. Looks like there is no easy solution. I stopped working on Signal for now. Best regards, Andreas.
